app/controllers/api/v1/plans_controller.rb

```diff
@@ -31,7 +38,14 @@ def destroy
         result = ::Plans::PrepareDestroyService.call(plan:)
 
         if result.success?
-          render_plan(reload_plan(code: result.plan.code)) # Reload to eager-load relationships
+          # Reload to eager-load relationships, like :entitlements
+          plan = Plan.includes(
```

Comment:
I still feel like we need to keep discarded ones here:

```suggestion
          plan = Plan.with_discarded.includes(
```